GiniVision/Classes/Core/CameraViewController.swift

```diff
@@ -614,15 +651,56 @@ extension CameraViewController {
     }
 }
 
+// MARK: - DocumentPickerCoordinatorDelegate
+
+extension CameraViewController: DocumentPickerCoordinatorDelegate {
+
+    func documentPicker(_ coordinator: DocumentPickerCoordinator,
+                        didPick documents: [GiniVisionDocument],
+                        from: DocumentPickerType,
+                        completion: DocumentPickerCompletion?) {
+        let loadingView = addValidationLoadingView()
+        self.validate(importedDocuments: documents) { validatedDocuments in
+            loadingView.removeFromSuperview()
+            let elementsWithError = validatedDocuments.filter { $0.1 != nil}
+            if let firstElement = elementsWithError.first,
+                let error = firstElement.1,
+                (!self.giniConfiguration.multipageEnabled || firstElement.0.type != .image) {
```

Comment:
Why is that? I really don't understand? Why only if the type is not an image and there's no multipage?

Also, it's so many conditions that my head hurts.

Reply:
I explained it above. But yes, too many conditions. I will try to simplify it.